src/internet/podcasts/podcastepisode.cpp

```diff
@@ -146,7 +146,16 @@ void PodcastEpisode::InitFromQuery(const QSqlQuery& query) {
   d->duration_secs_ = query.value(6).toInt();
   d->url_ = QUrl::fromEncoded(query.value(7).toByteArray());
   d->listened_ = query.value(8).toBool();
-  d->listened_date_ = QDateTime::fromTime_t(query.value(9).toUInt());
+
+  // After setting QDateTime to invalid state, it's saved into database as time_t,
+  // when this number(4294967295) is read back from database, it creates a valid QDateTime.
+  // So to make it behave consistently, this change is needed.
+  if (query.value(9).toUInt() == 4294967295) {
```

Comment:
Maybe this would be better as a separate PR?  And also use std::numeric_limits<uint32_t>::max() - 1 or something.